src/main/java/fi/metatavu/metamind/bot/slots/LocalizedNumberSlot.java

```diff
@@ -0,0 +1,172 @@
+        return result;
+      }
+    }
+
+    return null;
+  }
+  
+  private Number englishWordStringToNumber(String token) {
+    return wordStringToNumber(token);
+  }
+  
+  @SuppressWarnings("squid:S3776") //It is complex
+  private Number finnishWordStringToNumber(String token) {
+    if (token == null || token.length() < 1)
+    {
+      return null;
+    }
+
+    token = token.toLowerCase().replaceAll("[\\s+\\-,]", "").trim();
+
+    long result = 0;
+    long finalResult = 0;
+    int tokenCount = 0;
+    
+    while (token.length() > 0 && tokenCount < MAX_TOKENS) {
+      tokenCount++;
+      String foundToken = null;
+
+      if (token.startsWith("miljoonaa")) {
+        result *= 1000000;
+        finalResult += result;
+        result = 0;
+        foundToken = "miljoonaa";
+      } else if (token.startsWith("miljoona")) {
+        finalResult += 1000000;
+        result = 0;
+        foundToken = "miljoona";
+      } else if (token.startsWith("tuhatta")) {
+        result *= 1000;
+        finalResult += result;
+        result = 0;
+        foundToken = "tuhatta";
+      } else if (token.startsWith("tuhat")) {
+        result += 1000;
+        foundToken = "tuhat";
+      } else if (token.startsWith("sataa")) {
+        result *= 100;
+        finalResult += result;
+        result = 0;
+        foundToken = "sataa";
+      } else if (token.startsWith("sata")) {
+        result += 100;
+        foundToken = "sata";
+      } else if (token.startsWith("kymmentä")) {
+        result *= 10;
+        finalResult += result;
+        result = 0;
+        foundToken = "kymmentä";
+      } else if (token.startsWith("toista")) {
+        result += 10;
+        foundToken = "toista";
+      } else if (token.startsWith("kymmenen")) {
+        result += 10;
+        foundToken = "kymmenen";
+      } else if (token.startsWith("yhdeksän")) {
+        result += 9;
+        foundToken = "yhdeksän";
+      } else if (token.startsWith("kahdeksan")) {
+        result += 8;
+        foundToken = "kahdeksan";
+      } else if (token.startsWith("seitsemän")) {
+        result += 7;
+        foundToken = "seitsemän";
+      } else if (token.startsWith("kuusi")) {
+        result += 6;
+        foundToken = "kuusi";
+      } else if (token.startsWith("viisi")) {
+        result += 5;
+        foundToken = "viisi";
+      } else if (token.startsWith("neljä")) {
+        result += 4;
+        foundToken = "neljä";
+      } else if (token.startsWith("kolme")) {
+        result += 3;
+        foundToken = "kolme";
+      } else if (token.startsWith("kaksi")) {
+        result += 2;
+        foundToken = "kaksi";
+      } else if (token.startsWith("yksi")) {
+        result += 1;
+        foundToken = "yksi";
+      } else if (token.startsWith("nolla")) {
+        result += 0;
+        foundToken = "nolla";
+      } else {
+        return null;
+      }
+      
+      token = token.replaceFirst(foundToken, "");
+    }
+    
+    finalResult += result;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this useless assignment to local variable "result". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1854)